docs/tutorials/custom_experiment.rst

```diff
@@ -0,0 +1,362 @@
+building data processor workflows to help with advanced analysis of
+experiment data.
+
+Subclassing an Existing Experiment
+----------------------------------
+
+Let's walk through the process of subclassing an existing experiment in the Qiskit
+Experiments library.
+
+The FineAmplitude Experiment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``FineAmplitude`` calibration experiment optimizes gate amplitude by repeating the 
+gate pulse N times, hence amplifying the under- or over-rotations.
+This experiment can be performed for a variety of rotations, and subclasses are 
+provided for the :math:`\pi` and :math:`\frac{\pi}{2}` rotations as ``FineXAmplitude`` and ``FineSXAmplitude`` respectively.
+These provided subclasses focus on the 0 <-> 1 transition, however, this experiment can also be performed for higher order transitions.
+
+Our objective is to create a new class, ``HigherOrderFineXAmplitude``, which calibrates 
+schedules on transitions other than the 0 <-> 1 transition for the :math:`\pi` rotation.
+In order to do this, we need to create a subclass as shown below.
+
+.. code-block::
+   
+    class HigherOrderFineXAmplitude(FineXAmplitude):
+        def _pre_circuit(self) -> QuantumCircuit:
+            """Return a preparation circuit.
+            
+            This method can be overridden by subclasses e.g. to calibrate schedules on
+            transitions other than the 0 <-> 1 transition.
+            """
+            circuit = QuantumCircuit(1)
+
+            circuit.x(0)
+
+            if self.experiment_options.add_sx:
+                circuit.sx(0)
+
+            if self.experiment_options.sx_schedule is not None:
+                sx_schedule = self.experiment_options.sx_schedule
+                circuit.add_calibration("sx", (self.physical_qubits[0],), sx_schedule, params=[])
+                circuit.barrier()
+
+            return circuit
+
+In this subclass we have overridden the ``_pre_circuit`` method in order to calibrate on higher energy transitions by using an initial X gate to populate the first excited state.
+
+Using the Subclass
+------------------
+
+Now, we can use our new subclass as we would the original parent class.
+Shown below are results from following the :ref:`fine-amplitude-cal` tutorial
+for detecting an over-rotated pulse using our new 
+``HigherOrderFineXAmplitude`` class in place of the original 
+:class:`.FineXAmplitude` class.
+You can try this for yourself and verify that your results are similar.
+
+.. code-block::
+   
+   DbAnalysisResultV1
+   - name: d_theta
+   - value: -0.020710672666906425 ± 0.0012903658449026907
+   - χ²: 0.7819653845899581
+   - quality: good
+   - device_components: ['Q0']
+   - verified: False
+
+Writing a custom experiment
+---------------------------
+
+Now we'll use what we've learned so far to make a full custom experiment from
+the :class:`.BaseExperiment` template.
+
+A randomized measurement experiment
```

Comment:
Please revise the code of the experiment and analysis classes, in addition to my comments. It doesn't have the form of an example code that helps learn how to write an experiment.

Reply:
I've started to break up the code with explanations in-between in the newest commit. What do you mean by this doesn't have the form of an example code? Do you think a dummy template experiment with the attributes, something like

```python
class CustomExperiment(BaseExperiment):
    def __init__(self, physical_qubits=None, dummy_option=None, backend=None):
        """Initialize the experiment."""
        super().__init__(physical_qubits, analysis=CustomAnalysis(),backend=backend)
        self.set_experiment_options(
            dummy_option=dummy_option
        )
    @classmethod
    def _default_experiment_options(cls) -> Options:
        options = super()._default_experiment_options()
        options.update_options(
            dummy_option=None,
        )
        return options
    ...
```

would be helpful before introducing this experiment?